Automattic/wp-calypso
# Refactor Jetpack and WPCom plan types

### Changes proposed in this Pull Request

This PR restructures the plan types defined in the `calypso-products` package, and updates the codebase accordingly.

### Implementation notes

- Refactored types in `packages/calypso-products/src/types.ts`. Mainly extracted the `WPComPlan` and `JetpackPlan` interface from the more generic `Plan`.
- Made sure optional methods such as `getAudience` are called as such.
- Removed deprecated argument in calls to `getPlanCompareFeatures `, `getShortDescription`, `getDescription`, `getBlogSignupFeatures`, `getPortfolioSignupFeatures`.
- Renamed `JetpackPurchasableItem` to `JetpackPurchasableItemSlug` for clarity.
- Renamed `ProductTranslations` to `JetpackProduct` for clarity.
- Replaced `JetpackPlanCardFeature` deprecated type by the `string` type.
- Added `TYPES_LIST` in `packages/calypso-products/src/constants/types.ts`
- Added `WPCOM_PRODUCTS` and `WPCOM_PLANS` in `packages/calypso-products/src/constants/wpcom.ts`
- Clean plans in `packages/calypso-products/src/plans-list.js` according to the aforementioned updates.

### Testing instructions

- Review code carefully, following the implementation notes along
- Download the PR and run Calypso
- Open the console
- Visit the plans page with both a simple and a Jetpack site
- Check that the page looks and behaves as in production
- Verify that you can complete a purchase
- Check that there's no relevant error in the console
